.github/workflows/smoketest-nrfconnect.yaml

```
@@ -0,0 +1,77 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Smoke test - nRF Connect SDK
+
+on:
+    pull_request:
+    workflow_dispatch:
+
+concurrency:
+    group: smoke-${{ github.ref }}-${{ github.workflow }}-${{ (github.event_name == 'pull_request' && github.event.number) || (github.event_name == 'workflow_dispatch' && github.run_number) || github.sha }}
+    cancel-in-progress: true
+
+jobs:
+    nrfconnect:
+        name: Run
+        timeout-minutes: 30
+        runs-on: ubuntu-latest
+        if: github.actor != 'restyled-io[bot]'
+
+        container:
+            image: connectedhomeip/chip-build-nrf-platform:0.5.56
+            volumes:
+                - "/tmp/bloat_reports:/tmp/bloat_reports"
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              with:
+                  submodules: true
+
+            - name: Set up environment for size reports
+              if: ${{ !env.ACT }}
+              env:
+                  GH_CONTEXT: ${{ toJson(github) }}
+              run: scripts/tools/memory/gh_sizes_environment.py "${GH_CONTEXT}"
+
+            - name: Bootstrap
+              timeout-minutes: 25
+              run: scripts/build/gn_bootstrap.sh
+            - name: Uploading bootstrap logs
+              uses: actions/upload-artifact@v2
+              if: ${{ always() }} && ${{ !env.ACT }}
+              with:
+                  name: bootstrap-logs
+                  path: |
+                   .environment/gn_out/.ninja_log
+                   .environment/pigweed-venv/*.log
+            - name: Update nRF Connect SDK revision to the currently recommended.
+              timeout-minutes: 10
+              run: scripts/run_in_build_env.sh "python3 scripts/setup/nrfconnect/update_ncs.py --update --shallow"
+            - name: Build example nRF Connect SDK Lock App on nRF52840 DK
+              timeout-minutes: 10
+              run: |
+                  scripts/examples/nrfconnect_example.sh lock-app nrf52840dk_nrf52840
```

```suggestion
                  scripts/examples/nrfconnect_example.sh all-clusters-app nrf52840dk_nrf52840 -DBUILD_WITH_DFU=1
```
I agree with @mrjerryjohns that all-clusters-app seems the best choice if we want to build once app per platform. Also, let's build it with the DFU support to additionally increase the coverage.

---

Done the change. DFU seems new - the full build does not do it either. I wonder if the full build should also have these flags.

---

In the full build, we build lighting-app with the DFU (by default), so we haven't enabled it for all-clusters-app. But we can consider changing that, too.